docs/Architecture Decision Record/011-buildpacks.md

```diff
@@ -33,3 +33,7 @@ Our recommendation is to move to Cloud.gov buildpacks at this time. They are alr
 **Notes**
 - Docker containers will still need to be maintained for local development and CI/CD
 - Docker containers will still need to be hardened for CI/CD
+
+## Restaging for updated buildpacks
+
+As described in #1045, cloud.gov will inform us that buildpacks we use have been updated to a newer version via e-mail to all users with their 'developer' role. The e-mail informs as to specific CloudFoundry CLI steps needed to restage the our applications with this newer version of our buildpack base. As we have already captured our deployment strategy process/commands in scripts/deploy-backend.sh, running that script is the only step necessary when our team receives this e-mail. 
```

Comment:
Did we want to start a document for a buildpack change log so we can trigger these automatically?

Reply:
I guess I worded that vaguely, by trigger these automatically I was referring to triggering the re-deploy automatically - which will happen on any merge to raft-tdp-main (at least to staging). My suggestion was to create a markdown document that we can update anytime one of these releases is done and then on merging the PR we would get the new buildpacks. It doesn't need to be anything fancy, something like this would probably do:
```
## Buildpacks Changelog
- 07/13/2021 [nodejs-buildpack v1.7.56](https://github.com/cloudfoundry/nodejs-buildpack/releases/tag/v1.7.56)
```
Then each time we get that email we can just add a new line to that MD file and open a PR to trigger to re-deploy.

If we want to err on the side of caution we can apply the changes ahead of these PRs to the dev environment using the restage commands as stated in the email since we don't care about downtime there. Pasting the commands here for reference:
```
  cf target -o hhs-acf-prototyping -s tanf-dev ; cf restage tdp-frontend-sandbox

  cf target -o hhs-acf-prototyping -s tanf-dev ; cf restage tdp-frontend-a11y

  cf target -o hhs-acf-prototyping -s tanf-dev ; cf restage tdp-frontend-qasp

  cf target -o hhs-acf-prototyping -s tanf-dev ; cf restage tdp-frontend-raft
```

Reply:
We discussed offline if I recall, and I like your solution however with fresh eyes I'm realizing that the e-mail doesn't tell us which buildpack got updated. I sent an e-mail to their support to see about having any information to put into a changelog. Should we pause this discussion until I hear back?

Reply:
Ideally they would put that information in the email but I'm not sure if that's something they can do without code changes so I don't think we could realistically . We only use two buildpacks though (https://github.com/cloudfoundry/python-buildpack and https://github.com/cloudfoundry/nginx-buildpack) so if we list the current versions in the initial document for both of them we can pretty easily deduce which one was updated by comparing the versions between dev and the document after running those restage commands above.

So the full flow could be:
- get email from cloud.gov
- run restage commands against dev
- get new buildpack versions from dev environment after restage
- open a PR to note version changes in buildpacks changelog
- merge PR so rolling deploy happens and issues the buildpack update to staging & (eventually) prod without downtime

Reply:
If we know it will only be those two, we can always browse their changelogs directly:
**https://github.com/cloudfoundry/nginx-buildpack/blob/master/CHANGELOG**
**https://github.com/cloudfoundry/python-buildpack/blob/master/CHANGELOG**

And just update our changelog file and use the PR for restaging both dev and staging. Ok, yeah, I will rewrite the document to capture this.